### PR TITLE
[11.0] [MIG] Hotel

### PR DESCRIPTION
**Pull request by serpentcs-dev1**

### CODE REVIEW

File: hotel/README.rst [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+* **Different Reports are also provided, mainly for hotel statistics**
+
+
+Usage
+=====
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/157/10.0
```
Comment (JayVora-SerpentCS) [inline hotel/README.rst]: 11.0

File: hotel/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# -*- coding: utf-8 -*-
+# See LICENSE file for full copyright and licensing details.
+
+{
+    'name': 'Hotel Management Base',
+    'version': '10.0.1.0.0',
```
Comment (JayVora-SerpentCS) [inline hotel/__manifest__.py]: 11.0.1.0.1

File: hotel/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    'author': 'Odoo Community Association (OCA), Serpent Consulting\
+                Services Pvt. Ltd., Odoo S.A.',
+    'category': 'Generic Modules/Hotel Management',
+    'license': 'AGPL-3',
+    'depends': ['sale_stock', 'point_of_sale', 'report'],
```
Comment (JayVora-SerpentCS) [inline hotel/__manifest__.py]: report does not exist in v11 anymore.

**Comment by serpentcs-dev1** [comment]
@JayVora-SerpentCS The migration commit is still needed to be added. This is the version 10 preserved commit code.

We will add the new code soon! Please review on the base of that.

**Comment by serpentcs-dev1** [comment]
@pedrobaeza We have preserved the history of V10 Commits here!

LGTM when comparing it to https://github.com/OCA/vertical-hotel/commits/10.0/hotel

Now, @KrishnaPrajapati-SerpentCS Will add the newly migrated V11 code on top of this!

cc @JayVora-SerpentCS

File: hotel/__manifest__.py [hunk 4]
```diff
@@ -1,14 +1,13 @@
...
 # See LICENSE file for full copyright and licensing details.
 
 {
-    'name': 'Hotel Management Base',
-    'version': '10.0.1.0.0',
-    'author': 'Odoo Community Association (OCA), Serpent Consulting\
-                Services Pvt. Ltd., Odoo S.A.',
+    'name': 'Hotel Management',
+    'version': '11.0.1.0.0',
+    'author': 'Serpent Consulting Services Pvt. Ltd., OpenERP SA',
```
Comment (serpentcs-dev1) [inline hotel/__manifest__.py]: @KrishnaPrajapati-SerpentCS Please keep OCA in the authors.

**Comment by gustavovalverde** [comment]
I'd like to help here on the migration, what are you guys needing?
- Code Review
- Functional Review
- Other?

**Comment by serpentcs-dev1** [comment]
@gustavovalverde Yes Gustavo, you are welcome, we need Code and Functional Review both.

So we could optimize it more and make it OCA compliant and ready to merge!

**Comment by omarouhari** [comment]
what remains to be done in migration and how can we validate changes?

**Comment by serpentcs-dev1** [comment]
@omarouhari The migration has been done, now we need reviews to improve the code quality.

To compare the v10.0 you can check it here : https://github.com/OCA/vertical-hotel/tree/10.0/hotel

**Comment by omarouhari** [comment]
@gustavovalverde I ask le you can accelerate the code review ?
thanks
@darshan-serpent I can't get the log of last build !! how can I do ?

**Comment by gustavovalverde** [comment]
I'll be working on it today.

**Comment by serpentcs-dev1** [comment]
> @darshan-serpent I can't get the log of last build !! how can I do ?

@omarouhari Travis build does not seems to running, Maybe @pedrobaeza Can help here.

**Comment by serpentcs-dev1** [comment]
@omarouhari @gustavovalverde Can you please Code/Functional Review this, so we could merge this and move ahead.

cc @JayVora-SerpentCS

File: hotel/security/hotel_security.xml [hunk 5]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!DOCTYPE xml>
+<odoo>
+
+    <data noupdate="1">
```
Comment (pntabuye) [inline hotel/security/hotel_security.xml]: data tag is deprecated. You can now put the  noupdate="1" next to odoo tag.

Comment (serpentcs-dev1) [reply hotel/security/hotel_security.xml]: Yes right indeed. Will amend that, thanks!

@pntabuye Also, can you review the Code functionally and technically of other PRs.

Comment (pntabuye) [reply hotel/security/hotel_security.xml]: I'll start checking tonight @darshan-serpent - I need this one for a client I am upgrading to Odoo 11.

**Comment by serpentcs-dev1** [comment]
@omarouhari @gustavovalverde Can you please Code/Functional Review this, so we could merge this and move ahead.

File: hotel/wizard/sale_make_invoice_advance.py [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        ctx = self.env.context.copy()
+        if self._context.get('active_model') == 'hotel.folio':
+            hotel_fol = self.env['hotel.folio']
+            hotel = hotel_fol.browse(self._context.get('active_ids',
+                                                       []))
+            ctx.update({'active_ids': [hotel.order_id.id],
+                        'active_id': hotel.order_id.id})
+        return super(SaleAdvancePaymentInv,
+                     self.with_context(ctx))._get_advance_payment_method()
+    advance_payment_method = fields.Selection([('delivered',
```
Comment (nikul-serpentcs) [inline hotel/wizard/sale_make_invoice_advance.py]: IMHO it's not a readable format.

ex. state = fields.Selection([...], required=True)

File: hotel/report/hotel_report.py [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+    @api.model
+    def get_report_values(self, docids, data):
+        self.model = self.env.context.get('active_model')
+        if data is None:
+            data = {}
+        if not docids:
+            docids = data['form'].get('docids')
+        folio_profile = self.env['hotel.folio'].browse(docids)
+        date_start = data['form'].get('date_start', fields.Date.today())
+        date_end = data['form'].get('date_end', str(datetime.now() +
```
Comment (nikul-serpentcs) [inline hotel/report/hotel_report.py]: IMHO it's not a readable format.

**Comment by pntabuye** [comment]
Code and Functional review done. Nothing to point out on my side. Good job porting the thing to 11.0 @nikul-serpentcs

**Comment by nikul-serpentcs** [comment]
@pntabuye Thank you for code & functional review!

@pedrobaeza  If code & functional reviews were done then Could you please move Forward.

**Comment by pedrobaeza** [comment]
There were some improvements that @DarioLodeiros wanted to do on this migration. Would it be possible to be done, Darío?

**Comment by YogeshMahera-SerpentCS** [comment]
@DarioLodeiros Could you please do needful?

**Comment by DarioLodeiros** [comment]
Hi! I just got back from holiday :)
We have all work on this repo: [https://github.com/hootel/hootel](url)
The v11 is WIP, but I think that the basic design is quite incompatible with the OCA hotel vertical. Anyway, you can view the code of the calendar reservation in v11 here (maybe, it would be the part that most interest you to incorporate): https://github.com/hootel/hootel/tree/hotel_calendar/hotel_calendar, if you can view it in action: https://drive.google.com/file/d/186dt2cEsX0ZlO2tVjlb8M3q7RS6d6zm8/view?usp=sharing
I would be happy to incorporate this work to the OCA in v11 or future versions, but the mix with the current OCA Hotel module I think is not compatible from the specifications that we have detected. 
Feel free to reuse all the code to improve your version, and when we finish v11, hopefully it can be incorporated in some way in the OCA, although unfortunately there is overlapping work with different perspectives between the two projects.

**Comment by JayVora-SerpentCS** [comment]
@DarioLodeiros You may make PRs one by one. We ourselves also keep improving the product by adding features, migrating to newer version.

**Comment by DarioLodeiros** [comment]
Hi jay! I cant adapt our code to make a compatible pr with this module, there are too many differences in the design of the models. I think that the pr would be disruptive, and we can propose a new OCA PR when we finish de v11 version... while, I can only put at your disposal the code if you think that the effort to adapt it is worth it.
Any other idea is welcome :)

**Comment by JayVora-SerpentCS** [comment]
noted, in that case, we should work as a team to mix and merge the ideas from both ends. Open to hear. May be you can use https://github.com/OCA/vertical-hotel/wiki and https://github.com/OCA/vertical-hotel/issues to share the designs you have.

**Comment by JayVora-SerpentCS** [comment]
Closing as of now, to move further with v12 plan of OCA days. We can always make new PR to fix whats missing.

**Pull request merged**